snuba/split.py

```diff
@@ -9,86 +9,155 @@
 # queries before hitting the 90d limit (2+20+200+2000 hours == 92 days).
 STEP_GROWTH = 10
 
+
 def split_query(query_func):
-    """
-    If a query is:
-        - ORDER BY timestamp DESC
-        - has no grouping
-        - has an offset/limit
-        - has a large time range
-    We know we have to reverse-sort the entire set of rows to return the small
-    chunk at the end of the time range, so optimistically split the time range
-    into smaller increments, and start with the last one, so that we can potentially
-    avoid querying the entire range.
-    """
     def wrapper(*args, **kwargs):
         body = args[0]
-        use_split, date_align, split_step = state.get_configs([
+        use_split = state.get_configs([
             ('use_split', 0),
-            ('date_align_seconds', 1),
-            ('split_step', 3600),  # default 1 hour
         ])
-        to_date = util.parse_datetime(body['to_date'], date_align)
-        from_date = util.parse_datetime(body['from_date'], date_align)
         limit = body.get('limit', 0)
         remaining_offset = body.get('offset', 0)
         orderby = util.to_list(body.get('orderby'))
 
+        total_col_count = len(util.all_referenced_columns(body))
+        min_col_query = dict(body, **{'selected_columns': ['project_id', 'event_id', 'timestamp']})
```

Comment:
Also, since this isn't used again (and doesn't necessarily generate a valid query), it seems to me that it'd make sense to restructure this so that `len(util.all_referenced_columns({**body, selected_columns=['project_id', 'event_id', 'timestamp']}))` is only actually evaluated if the other conditionals starting on line 28 are `True`.